src/embedding/ad.ts

```diff
@@ -0,0 +1,417 @@
+    title?: string;
+    format?: 'xlsx' | 'csv' | 'raw';
+    mergeHeaders?: boolean;
+}
+
+// NOTE: AFM is ommitted on purpose; it should be added by AD itself; create new type using Omit<>
+export interface IExportConfig extends IBaseExportConfig {
+    showFilters?: boolean;
+}
+
+export namespace EmbeddedAnalyticalDesigner {
+    export type IADMessageEvent<T, TBody> = IGdcMessageEvent<GdcProductName.ANALYTICAL_DESIGNER, T, TBody>;
+    export type IAdGdcMessageEventDataWrapper<T, TBody> = IGdcMessageEventDataWrapper<
+        GdcProductName.ANALYTICAL_DESIGNER,
+        T,
+        TBody
+    >;
+
+    export enum GdcAdCommandType {
+        DrillableItems = 'drillableItems',
+        OpenInsight = 'openInsight',
+        Save = 'saveInsight',
+        SaveAs = 'saveAsInsight',
+        Export = 'exportInsight',
+        Undo = 'undo',
+        Redo = 'redo'
+    }
+
+    export enum GdcAdEventType {
+        ListeningForDrillableItems = 'listeningForDrillableItems',
+        NewInsightInitialized = 'newInsightInitialized',
+        InsightOpened = 'insightOpened',
+        // use `visualizationSaved` because of backward compatibility
+        InsightSaved = 'visualizationSaved',
+        UndoFinished = 'undoFinished',
+        RedoFinished = 'redoFinished',
+        ExportFinished = 'exportInsightFinished'
+    }
+
+    /**
+     * This event will be emitted if AD runs into errors while processing the posted command.
+     *
+     * @remarks see {@link GdcErrorType} for types of errors that may fly
+     */
+    export type AdCommandFailed = CommandFailed<GdcProductName.ANALYTICAL_DESIGNER>;
+
+    export type AdCommandFailedData = CommandFailedData<GdcProductName.ANALYTICAL_DESIGNER>;
+
+    export function isAdCommandFailedData(obj: any): obj is AdCommandFailedData {
+        return isCommandFailedData<GdcProductName.ANALYTICAL_DESIGNER>(obj);
+    }
+
+    //
+    // Drillable Items command
+    //
+
+    export interface ISimpleDrillableItemsCommandBody {
+        uris?: string[];
+        identifiers?: string[];
+    }
+
+    export interface IDrillableItemsCommandBody extends ISimpleDrillableItemsCommandBody {
+        composedFrom?: ISimpleDrillableItemsCommandBody;
+    }
+
+    export type DrillableItemsCommand = IADMessageEvent<GdcAdCommandType.DrillableItems, IDrillableItemsCommandBody>;
+
+    export type DrillableItemsCommandData = IAdGdcMessageEventDataWrapper<
+        GdcAdCommandType.DrillableItems,
+        IDrillableItemsCommandBody
+    >;
+
+    /**
+     * Type-guard checking whether object is an instance of DrillableItemsCommandData.
+     *
+     * @param obj - object to test
+     */
+    export function isDrillableItemsCommandData(obj: any): obj is DrillableItemsCommandData {
+        return get(obj, 'gdc.event.name') === GdcAdCommandType.DrillableItems;
+    }
+
+    //
+    // Open insight command
+    //
+
+    export interface IOpenInsightCommandBody {
+        dataset?: string | null;
```

Comment:
why are we doing optional prop with possibly nullable value? what case does this address?

also, what's the purpose of the additional properties (aside from reportId?).. wouldn't reportId be enough?